include/GafferScene/Instancer.h

```diff
@@ -53,15 +53,31 @@ class GAFFERSCENE_API Instancer : public BranchCreator
 
 		GAFFER_GRAPHCOMPONENT_DECLARE_TYPE( GafferScene::Instancer, InstancerTypeId, BranchCreator );
 
+		enum PrototypeMode
+		{
+			IndexedRootsList = 0,
+			IndexedRootsVariable,
+			RootPerVertex,
+		};
+
 		Gaffer::StringPlug *namePlug();
 		const Gaffer::StringPlug *namePlug() const;
 
 		ScenePlug *prototypesPlug();
 		const ScenePlug *prototypesPlug() const;
 
+		Gaffer::IntPlug *prototypeModePlug();
+		const Gaffer::IntPlug *prototypeModePlug() const;
+
 		Gaffer::StringPlug *prototypeIndexPlug();
 		const Gaffer::StringPlug *prototypeIndexPlug() const;
 
+		Gaffer::StringPlug *prototypeRootsPlug();
+		const Gaffer::StringPlug *prototypeRootsPlug() const;
+
+		Gaffer::StringVectorDataPlug *prototypeRootPathsPlug();
+		const Gaffer::StringVectorDataPlug *prototypeRootPathsPlug() const;
+
```

Comment:
I find the `prototypeRoots` vs `prototypeRootPaths` distinction a bit hard to keep clear in my head. Any thoughts on alternative namings? Perhaps `prototypesRootsList` and `prototypeRootsVariable`, to tie them directly to the corresponding enum values?

Reply:
In the code I don't mind at all, those new names are fine. But in the UI... switch to IndexedRootsVaraible mode and we have 2 plugs for the user to set, `prototypeIndex` and `prototypeRootsVariable`, both of which are primvar names.. it feels a bit awkward.

Reply:
I agree it's a bit awkward to have 2 plugs, but only one suffixed with "Variable". But I think there's something to be said for the explicitness of "RootsList" vs "RootsVariable". Could maybe use a metadata label for the UI? Not sure if by trying to fix one problem I'm just creating another here...

Reply:
Yeah, I really don't like labels masking the true plug names... would it be any better to call them `prototypeRoots` and `prototypeRootsList`? Or is that still too ambiguous?

Reply:
I should explain where I found this confusing, in case it helps at all. In the UI it was clear because of the activators. And when reviewing the Instancer internals it was pretty clear too. It was when following the unit tests I found things less clear. When setting up an Instancer programmatically, without the benefit of UI activators, it was hard to keep in my head which plugs mattered at any given time.

> Yeah, I really don't like labels masking the true plug names... would it be any better to call them prototypeRoots and prototypeRootsList? Or is that still too ambiguous?

So we're just renaming `prototypeRootPaths->prototypeRootsList`? That feels like a worthwhile improvement to me, and doesn't seem to introduce any extra awkwardness.

Reply:
Addressed in 63aa83ae508a4c0e4c805a1bbcbc053257801fa5